lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,639 @@
+	 *
+	 * This configuration is serialized and sent to the client as part of the
+	 * interactivity data and can be accessed by the client interactivity stores.
+	 *
+	 * @since 6.5.0
+	 * @var array
+	 */
+	private $config_data = array();
+
+	/**
+	 * Gets and/or sets the initial state of an Interactivity API store for a
+	 * given namespace.
+	 *
+	 * If state for that store namespace already exists, it merges the new
+	 * provided state with the existing one.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $store_namespace The unique store namespace identifier.
+	 * @param array  $state           Optional. The array that will be merged with the existing state for the specified
+	 *                                store namespace.
+	 * @return array The current state for the specified store namespace.
+	 */
+	public function state( string $store_namespace, array $state = null ): array {
+		if ( ! isset( $this->state_data[ $store_namespace ] ) ) {
+			$this->state_data[ $store_namespace ] = array();
+		}
+		if ( is_array( $state ) ) {
+			$this->state_data[ $store_namespace ] = array_replace_recursive(
+				$this->state_data[ $store_namespace ],
+				$state
+			);
+		}
+		return $this->state_data[ $store_namespace ];
+	}
+
+	/**
+	 * Gets and/or sets the configuration of the Interactivity API for a given
+	 * store namespace.
+	 *
+	 * If configuration for that store namespace exists, it merges the new
+	 * provided configuration with the existing one.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $store_namespace The unique store namespace identifier.
+	 * @param array  $config          Optional. The array that will be merged with the existing configuration for the
+	 *                                specified store namespace.
+	 * @return array The current configuration for the specified store namespace.
+	 */
+	public function config( string $store_namespace, array $config = null ): array {
+		if ( ! isset( $this->config_data[ $store_namespace ] ) ) {
+			$this->config_data[ $store_namespace ] = array();
+		}
+		if ( is_array( $config ) ) {
+			$this->config_data[ $store_namespace ] = array_replace_recursive(
+				$this->config_data[ $store_namespace ],
+				$config
+			);
+		}
+		return $this->config_data[ $store_namespace ];
+	}
+
+	/**
+	 * Prints the serialized client-side interactivity data.
+	 *
+	 * Encodes the config and initial state into JSON and prints them inside a
+	 * script tag of type "application/json". Once in the browser, the state will
+	 * be parsed and used to hydrate the client-side interactivity stores and the
+	 * configuration will be available using a `getConfig` utility.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_client_interactivity_data() {
+		if ( ! empty( $this->state_data ) || ! empty( $this->config_data ) ) {
+			wp_print_inline_script_tag(
+				wp_json_encode(
+					array(
+						'config' => (object) $this->config_data,
+						'state'  => (object) $this->state_data,
```

Comment:
Do we want the array keys if we don't have config_data but have state_data or viceversa?

Reply:
If not, we could ship something like this (code by @dmsnell )

```
if  ( ! ( $has_state_data || $has_config_data ) ) {
     return;
}

$store = array();

if ( $has_state_data ) {
    $store['state'] = (object) $this->state_data;
}

if ( $has_config_data ) {
   $store['config'] = (object) $this->config_data;
}

wp_print_inline_script_tag(
    wp_json_encode( $store, … )
)
 …
 ```